spec/search_spec.rb

```diff
@@ -0,0 +1,41 @@
+require_relative 'spec_helper'
+require_relative 'request_spec_helper'
+require_relative '../lib/chores/populator'
+require "queryparams"
+
+RSpec.describe "search" do
+  describe "/search" do
+    it "searches all fields by default" do
+      populate_manifests
+      search_for "around+the+corner"
+      expect(JSON.parse(last_response.body)["total"]).to eq 10
+    end
+
+    it "searches by specific field" do
+      populate_manifests
+      search_for "content.generator.name:generator"
+      expect(JSON.parse(last_response.body)["total"]).to eq 10
+    end
+
+    it "searched by structured query" do
+      populate_manifests
+      search_for_advanced({ "content.generator.name" => "generator" })
+      expect(JSON.parse(last_response.body)["total"]).to eq 10
+    end
+  end
+
+  def populate_manifests
+    populator = Populator.new(Manifest)
+    populator.run(10)
```

Comment:
I anticipate needing to write tests for sorting and ranking, so having more than 2 by default would be useful.